### PR TITLE
Adding support for wire length units

### PR DESCRIPTION
**Pull request by zombielinux**
Will also be editing dataclasses.py to add the support as well as defaulting units to meters if unpopulated.

### CODE REVIEW

File: src/wireviz/Harness.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def create_graph(self) -> Graph:
                      f'{cable.wirecount}x' if cable.show_wirecount else None,
                      f'{cable.gauge} {cable.gauge_unit}{awg_fmt}' if cable.gauge else None,
                      '+ S' if cable.shield else None,
-                     f'{cable.length} m' if cable.length > 0 else None,
+                     f'{cable.length} {cable.lengthunit}' if cable.length > 0 else None,
```
Comment (kvid) [inline src/wireviz/Harness.py:173]: I can see an unfortunate difference in naming style between these two attribute pairs:
- `cable.gauge` and `cable.gauge_unit`
- `cable.length` and `cable.lengthunit`

Feel free to discuss naming conventions in a new issue, but I suggest we try to keep it consistent.

**Review by kvid — CHANGES_REQUESTED** [review]
- Just a friendly advice: Read the hints about writing good commit messages: https://github.com/formatc1702/WireViz/blob/feature/syntax-description/docs/CONTRIBUTING.md#hints
- Please include all commits needed for this feature into your `patch-1` branch and re-push it, i.e. join (rebase or cherry-pick) the #162 commit(s) into this PR. The automatic PR checks currently fail in this PR because the `DataClasses.py` changes are missing.

File: src/wireviz/Harness.py (hunk 1 above)
Comment (stevegt) [reply src/wireviz/Harness.py:173]: added the underbar to length_unit in #171

**Comment by stevegt** [comment]
merged with #162 and submitted as #171

**Comment by formatc1702** [comment]
Closed as part of #196.

**Pull request closed without merging**